rspace/src/main/scala/coop/rchain/rspace/history/HistoryRepositoryImpl.scala

```diff
@@ -149,6 +149,12 @@ final case class HistoryRepositoryImpl[F[_]: Sync: Par, C, P, A, K](
         s"${key.toHex};delete-join;0"
     }
 
+  private def transformAndStore(hotStoreActions: List[HotStoreAction]): F[List[HistoryAction]] =
+    for {
+      transformedActions <- Sync[F].delay(hotStoreActions.map(transform))
+      historyActions     <- storeLeaves(transformedActions)
```

Comment:
`hotStoreActions.map(transform)` is pure, so it would be enough to delay only `storeLeaves` here.

Reply:
that's what's been here previously, and it ended up being executed during computation definition (becuase it's pure). Turns out parallelism is effectful :D

Reply:
Interesting find!